tests/copilot_helper/test_command_pipeline.py

```diff
@@ -45,39 +46,95 @@ def test_pipeline_generate_with_git_repo_creates_the_pipeline_configuration(
+    del pipelines["codebases"]
+    Path("pipelines.yml").write_text(yaml.dump(pipelines))
 
     CliRunner().invoke(generate)
 
-    expected_files_dir = Path(EXPECTED_FILES_DIR) / "pipeline" / "pipelines" / "environments"
-    assert_yaml_in_output_file_matches_expected(buildspec, expected_files_dir / "buildspec.yml")
-    assert_yaml_in_output_file_matches_expected(manifest, expected_files_dir / "manifest.yml")
-    assert_yaml_in_output_file_matches_expected(
-        cfn_patch, expected_files_dir / "overrides/cfn.patches.yml"
-    )
+    environments_files = setup_output_file_paths_for_environments(tmp_path)
+    for file in environments_files:
+        assert Path(file).exists()
+
+    codebase_files = setup_output_file_paths_for_codebases(tmp_path)
+    for file in codebase_files:
+        assert not Path(file).exists()
+
+
+@freeze_time("2023-08-22 16:00:00")
+@patch("dbt_copilot_helper.jinja2_tags.version", new=Mock(return_value="v0.1-TEST"))
+@patch("boto3.client")
+def test_pipeline_generate_with_only_codebases_creates_the_pipeline_configuration(
+    mocked_boto3_client, tmp_path, switch_to_tmp_dir_and_copy_fixtures
+):
+    mock_codestar_connections_boto_client(mocked_boto3_client, ["test-app"])
+    setup_git_repository()
+
+    pipelines = yaml.safe_load(Path("pipelines.yml").read_text())
+    del pipelines["environments"]
+    Path("pipelines.yml").write_text(yaml.dump(pipelines))
+
+    CliRunner().invoke(generate)
+
+    environments_files = setup_output_file_paths_for_environments(tmp_path)
+    for file in environments_files:
+        assert not Path(file).exists()
+
+    codebase_files = setup_output_file_paths_for_codebases(tmp_path)
+    for file in codebase_files:
+        assert Path(file).exists()
+
+
+@freeze_time("2023-08-22 16:00:00")
+@patch("dbt_copilot_helper.jinja2_tags.version", new=Mock(return_value="v0.1-TEST"))
+@patch("boto3.client")
+def test_pipeline_generate_with_empty_pipelines_yml_does_nothing(
+    mocked_boto3_client, tmp_path, switch_to_tmp_dir_and_copy_fixtures
+):
+    Path("pipelines.yml").write_text(yaml.dump({}))
+
+    result = CliRunner().invoke(generate)
+
+    assert "Error: No environment or codebase pipelines defined in pipelines.yml" in result.output
+
+
+@freeze_time("2023-08-22 16:00:00")
+@patch("dbt_copilot_helper.jinja2_tags.version", new=Mock(return_value="v0.1-TEST"))
+@patch("boto3.client")
+def test_pipeline_generate_overwrites_any_existing_config_files(
+    mocked_boto3_client, tmp_path, switch_to_tmp_dir_and_copy_fixtures
+):
+    mock_codestar_connections_boto_client(mocked_boto3_client, ["test-app"])
+    setup_git_repository()
+    environments_files = setup_output_file_paths_for_environments(tmp_path)
+    codebases_files = setup_output_file_paths_for_codebases(tmp_path)
+
+    result = CliRunner().invoke(generate)
+    for file in environments_files + codebases_files:
+        assert_file_created_in_stdout(file, result, tmp_path)
```

Comment:
I wonder if it's worth slipping an extra file in after this and asserting that it does not exist after the next generate?

Reply:
We've not implemented "clean up before adding files" as yet, so leaving this out.